photosynthesis_metrics/tv.py

```diff
@@ -98,33 +109,26 @@ def __init__(self, size_average: bool = True, reduction_type: str = 'l2', reduct
         self.reduction_type = reduction_type
         self.reduction = reduction
 
-    def forward(self, prediction: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
+    def forward(self, prediction: torch.Tensor) -> torch.Tensor:
         r"""Computation of Total Variation (TV) index as a loss function.
 
         Args:
             prediction: Tensor of prediction of the network.
-            target: Reference tensor.
 
         Returns:
             Value of TV loss to be minimized.
         """
-        prediction, target = _adjust_dimensions(x=prediction, y=target)
-        _validate_input(x=prediction, y=target)
+        prediction = _adjust_tensor_dimensions(prediction)
+        _validate_input(prediction)
```

Comment:
Let's change the order in order to validate input first and then adjust the dimensions. Current implementation makes `isinstance(x, torch.Tensor)` at  `_validate_input` pointless.

Reply:
yeah, it's a good point!